### PR TITLE
SoC: nrf53 and nrf91: fix partition sizes

### PR DESCRIPTION
**Pull request by ioannisg**
The work in https://github.com/zephyrproject-rtos/zephyr/pull/25874 has changed the starting addresses for the non-secure slot flash partitions for Nordic official DK boards with Cortex-M33 MCUs with TrustZone-m capability.

The new offsets of non-secure flash partitions do not comply with the requirements of the Nordic IDAU (SPU) and need a re-alignment. This PR fixes this for nRF53 and nRF91 (only the Nordic DK)

### CODE REVIEW

File: boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_partition_conf.dts [hunk 1]
```diff
@@ -22,19 +22,19 @@
  */
 
 &slot0_partition {
-	reg = <0x0000c000 0x30000>;
+	reg = <0x00010000 0x40000>;
 };
 
 &slot0_ns_partition {
-	reg = <0x0003e000 0x40000>;
+	reg = <0x00040000 0x30000>;
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_partition_conf.dts:29]:
```suggestion
	reg = <0x00050000 0x30000>;
```

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts [hunk 2]
```diff
@@ -22,19 +22,19 @@
  */
 
 &slot0_partition {
-	reg = <0x0000c000 0x30000>;
+	reg = <0x0000c000 0x40000>;
 };
 
 &slot0_ns_partition {
-	reg = <0x0003e000 0x40000>;
+	reg = <0x0003e000 0x30000>;
```
Comment (anangl) [inline boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts:29]:
```suggestion
	reg = <0x00050000 0x30000>;
```

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts [hunk 3]
```diff
@@ -22,19 +22,19 @@
...
 };
 
 &slot1_partition {
-	reg = <0x0007e000 0x30000>;
+	reg = <0x0007e000 0x40000>;
```
Comment (anangl) [inline boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts:33]:
```suggestion
	reg = <0x00080000 0x40000>;
```

File: boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts [hunk 4]
```diff
@@ -22,19 +22,19 @@
...
 };
 
 &slot1_ns_partition {
-	reg = <0x000b0000 0x40000>;
+	reg = <0x000b0000 0x30000>;
```
Comment (anangl) [inline boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_partition_conf.dts:37]:
```suggestion
	reg = <0x000c0000 0x30000>;
```

**Review by ioannisg — COMMENTED** [review]
@anangl updated, thanks
Does it make sense to leave MCUboot to be up to 0xc000, since slot_0 starts at 0x10000 anyways?

**Comment by anangl** [comment]
> @anangl updated, thanks
> Does it make sense to leave MCUboot to be up to 0xc000, since slot_0 starts at 0x10000 anyways?

I don't know, actually, seems like it doesn't. Perhaps @de-nordic or @nvlsianpu could comment on this?

**Comment by nvlsianpu** [comment]
^^ why not to make it 0x10000 of size then?

**Comment by ioannisg** [comment]
> ^^ why not to make it 0x10000 of size then?

That's contradicts your modifcation in #25874, so I wanted you and @de-nordic to ACK :)

**Comment by ioannisg** [comment]
> > ^^ why not to make it 0x10000 of size then?
> 
> That's contradicts your modifcation in #25874, so I wanted you and @de-nordic to ACK :)

@nvlsianpu, changed it now to be 0x10000 for both 53 and 91.

**Comment by de-nordic** [comment]
> 
> That's contradicts your modifcation in #25874, so I wanted you and @de-nordic to ACK :)

I am fine with it, it was basically configuration  change resulting from https://github.com/zephyrproject-rtos/zephyr/issues/25664, I just did it.

**Comment by de-nordic** [comment]
... and you are increasing the size of the boot partition so it does not hurt me :)

**Comment by ioannisg** [comment]
> ... and you are increasing the size of the boot partition so it does not hurt me :)

Yes, it is unfortunate that we need to have different sizes for the boot partition. Would be nice if nRF52 could have the same but i understand that several 52 SoCs have less flash size.

**Review by de-nordic — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**